Code/Framework/AzCore/Tests/EBus/TestEventProcessingPolicy.cpp

```diff
@@ -0,0 +1,173 @@
+            AZStd::invoke(AZStd::forward<Function>(func), AZStd::forward<Interface>(iface), AZStd::forward<InputArgs>(args)...);
+        }
+    };
+
+    struct ITestHookEvents
+    {
+        AZ_RTTI(ITestHookEvents, "{349C00E8-65C7-4FF6-B505-E047238618CC}");
+
+        virtual void Contribute(AZStd::vector<AZStd::string>& stack) const = 0;
+    };
+
+    struct TestHookEventTraits
+        : public AZ::EBusTraits
+    {
+        static const AZ::EBusHandlerPolicy HandlerPolicy = AZ::EBusHandlerPolicy::Multiple;
+        static const AZ::EBusAddressPolicy AddressPolicy = AZ::EBusAddressPolicy::Single;
+        using EventProcessingPolicy = TestEventProcessingPolicy;
+    };
+
+    using TestHookEventBus = AZ::EBus<ITestHookEvents, TestHookEventTraits>;
+
+    struct TestOneEventHook
+        : public TestHookEventBus::Handler
+    {
+        AZ_RTTI(TestOneEventHook, "{BC6F4FCD-F035-4527-9262-7582ACE33EDF}", TestHookEventBus::Handler);
+
+        TestOneEventHook()
+        {
+            TestHookEventBus::Handler::BusConnect();
+        }
+
+        ~TestOneEventHook()
+        {
+            TestHookEventBus::Handler::BusDisconnect();
+        }
+
+        void Contribute(AZStd::vector<AZStd::string>& stack) const override
+        {
+            stack.push_back("TestOneEventHook");
+        }
+
+        const char* Type()
+        {
+            return "TestOneEventHook";
+        }
+    };
+
+    struct TestTwoEventHook
+        : public TestHookEventBus::Handler
+    {
+        AZ_RTTI(TestTwoEventHook, "{E6C52A54-56DA-4BB7-BEA5-B1CF07640A31}", TestHookEventBus::Handler);
+
+        TestTwoEventHook()
+        {
+            TestHookEventBus::Handler::BusConnect();
+        }
+
+        ~TestTwoEventHook()
+        {
+            TestHookEventBus::Handler::BusDisconnect();
+        }
+
+        void Contribute(AZStd::vector<AZStd::string>& stack) const override
+        {
+            stack.push_back("TestTwoEventHook");
+        }
+
+        const char* Type()
+        {
+            return "TestTwoEventHook";
+        }
+    };
+
+    struct TestReporter final
+        : public ITestEventProcessingReporter
+    {
+        TestReporter()
+        {
+            AZ::Interface<ITestEventProcessingReporter>::Register(this);
+        }
+
+        ~TestReporter()
+        {
+            AZ::Interface<ITestEventProcessingReporter>::Unregister(this);
+        }
+
+        void Report(const char* item) override
+        {
+            m_reports.push_back(item);
+        }
+
+        AZStd::vector<AZStd::string> m_reports;
+    };
+
+    TEST_F(EventProcessingPolicyTests, SimplePolicy_Works)
+    {
+        TestOneEventHook testOneEventHook;
+        TestTwoEventHook testTwoEventHook;
+        TestReporter testReporter;
+
+        AZStd::vector<AZStd::string> data;
+        TestHookEventBus::Broadcast(&TestHookEventBus::Events::Contribute, data);
+
+        EXPECT_EQ(data.size(), 2);
+        EXPECT_EQ(testReporter.m_reports.size(), 2);
```

Comment:
When comparing conditions using `EXPECT_EQ` or `EXPECT_NE`, the expected value comes first and the actual value second.

If the test fails, googletest will print out a message such as "expected 2, actual <what is actually came out to>"

```suggestion
        EXPECT_EQ(2, data.size());
        EXPECT_EQ(2, testReporter.m_reports.size());
```